lib/application.js

```diff
@@ -156,9 +156,23 @@ module.exports = class Application extends Emitter {
   handleRequest(ctx, fnMiddleware) {
     const res = ctx.res;
     res.statusCode = 404;
-    const onerror = err => ctx.onerror(err);
-    const handleResponse = () => respond(ctx);
+
+    const responding = Symbol('responding');
+    this.once(responding, () => this.emit('respond', ctx));
+
+    const onerror = err => {
+      if (null != err) {
```

Comment:
the old code will execute `onerror` no matter the err is null or not. So I think this is a breaking change.

Reply:
@fengmk2 please give yourself a pleasure to check first line of `onerror`. What kind of code review is that?!!